src/box/memtx_engine.cc

```diff
@@ -278,6 +278,9 @@ memtx_engine_recover_snapshot(struct memtx_engine *memtx,
 			say_error("snapshot `%s' has no EOF marker", cursor.name);
 	}
 
+	if (is_space_system < 0)
+		panic("snapshot `%s' has no system spaces", cursor.name);
```

Comment:
`is_space_system` < 0 means that the snapshot contains no spaces at all. If it contains at least one space (system or user), `is_space_system` will be >= 0, and the check will pass. IMO this check makes no sense. Looks like the intention of commit 0be1243ecc8a0b56b079d82d4c713982a05382f0, which added the check, was to fail recovery if at least one system space is missing, but adding such a check would be complicated. Let's just drop it? While we are at it, we should change the type of `is_space_system` to bool.

TBH I find the aforementioned commit dubious - according to its description (and #5422), the intention was to enable the `force_recovery` option for snapshots, but it was in fact already enabled. The only useful thing the commit did was enable recovery of truncated snapshots (without EOF marker). For some reason, it also added some broken checks that disabled the `force_recovery` option for system spaces. Maybe, just revert that part of the commit?

Reply:
`force_recovery` disabling for system spaces actually works ok, [it was requested here](https://github.com/tarantool/tarantool/pull/5443#issuecomment-717833930), so let's keep it as is.